File: server/module/module-mall/mall-server/src/convert/mall_promotion_seckill_activity.rs
use sea_orm::{Set, NotSet};
use crate::model::mall_promotion_seckill_activity::{self, Model as MallPromotionSeckillActivity, ActiveModel as MallPromotionSeckillActivityActiveModel};
use mall_model::request::mall_promotion_seckill_activity::{CreateMallPromotionSeckillActivityRequest, UpdateMallPromotionSeckillActivityRequest};
use mall_model::response::mall_promotion_seckill_activity::MallPromotionSeckillActivityResponse;

pub fn create_request_to_model(request: &CreateMallPromotionSeckillActivityRequest) -> MallPromotionSeckillActivityActiveModel {
    MallPromotionSeckillActivityActiveModel {
        spu_id: Set(request.spu_id.clone()),
        name: Set(request.name.clone()),
        status: Set(request.status.clone()),
        remark: request.remark.as_ref().map_or(NotSet, |remark| Set(Some(remark.clone()))),
        start_time: Set(request.start_time.clone()),
        end_time: Set(request.end_time.clone()),
        sort: Set(request.sort.clone()),
        config_ids: Set(request.config_ids.clone()),
        total_limit_count: request.total_limit_count.as_ref().map_or(NotSet, |total_limit_count| Set(Some(total_limit_count.clone()))),
        single_limit_count: request.single_limit_count.as_ref().map_or(NotSet, |single_limit_count| Set(Some(single_limit_count.clone()))),
        stock: request.stock.as_ref().map_or(NotSet, |stock| Set(Some(stock.clone()))),
        total_stock: request.total_stock.as_ref().map_or(NotSet, |total_stock| Set(Some(total_stock.clone()))),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateMallPromotionSeckillActivityRequest, existing: MallPromotionSeckillActivity) -> MallPromotionSeckillActivityActiveModel {
    let mut active_model: MallPromotionSeckillActivityActiveModel = existing.into();
    if let Some(spu_id) = &request.spu_id { 
        active_model.spu_id = Set(spu_id.clone());
    }
    if let Some(name) = &request.name { 
        active_model.name = Set(name.clone());
    }
    if let Some(status) = &request.status { 
        active_model.status = Set(status.clone());
    }
    if let Some(remark) = &request.remark { 
        active_model.remark = Set(Some(remark.clone()));
    }
    if let Some(start_time) = &request.start_time { 
        active_model.start_time = Set(start_time.clone());
    }
    if let Some(end_time) = &request.end_time { 
        active_model.end_time = Set(end_time.clone());
    }
    if let Some(sort) = &request.sort { 
        active_model.sort = Set(sort.clone());
    }
    if let Some(config_ids) = &request.config_ids { 
        active_model.config_ids = Set(config_ids.clone());
    }
    if let Some(total_limit_count) = &request.total_limit_count { 
        active_model.total_limit_count = Set(Some(total_limit_count.clone()));
    }
    if let Some(single_limit_count) = &request.single_limit_count { 
        active_model.single_limit_count = Set(Some(single_limit_count.clone()));
    }
    if let Some(stock) = &request.stock { 
        active_model.stock = Set(Some(stock.clone()));
    }
    if let Some(total_stock) = &request.total_stock { 
        active_model.total_stock = Set(Some(total_stock.clone()));
    }
    active_model
}

pub fn model_to_response(model: MallPromotionSeckillActivity) -> MallPromotionSeckillActivityResponse {
    MallPromotionSeckillActivityResponse { 
        id: model.id,
        spu_id: model.spu_id,
        name: model.name,
        status: model.status,
        remark: model.remark,
        start_time: model.start_time,
        end_time: model.end_time,
        sort: model.sort,
        config_ids: model.config_ids,
        total_limit_count: model.total_limit_count,
        single_limit_count: model.single_limit_count,
        stock: model.stock,
        total_stock: model.total_stock,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}

File: admin/src/api/mall_promotion_seckill_activity.ts
import { PaginatedRequest, PaginatedResponse } from '@/base/page';
import { api } from '@/utils/request';

const apis = {
  create: '/mall/mall_promotion_seckill_activity/create', // 新增
  update: '/mall/mall_promotion_seckill_activity/update', // 修改
  delete: '/mall/mall_promotion_seckill_activity/delete', // 删除
  get: '/mall/mall_promotion_seckill_activity/get', // 单条查询
  list: '/mall/mall_promotion_seckill_activity/list', // 列表查询
  page: '/mall/mall_promotion_seckill_activity/page', // 分页查询
  enable: '/mall/mall_promotion_seckill_activity/enable', // 启用
  disable: '/mall/mall_promotion_seckill_activity/disable', // 禁用
}

export interface MallPromotionSeckillActivityRequest {
  id: number; // 秒杀活动编号
  spu_id: number; // 秒杀活动商品
  name: string; // 秒杀活动名称
  status: number; // 活动状态
  remark: string; // 备注
  start_time: string; // 活动开始时间
  end_time: string; // 活动结束时间
  sort: number; // 排序
  config_ids: string; // 秒杀时段 id 数组
  total_limit_count: number; // 总限购数量
  single_limit_count: number; // 单次限够数量
  stock: number; // 秒杀库存
  total_stock: number; // 秒杀总库存
  }

export interface MallPromotionSeckillActivityResponse {
  id: number; // 秒杀活动编号
  spu_id: number; // 秒杀活动商品
  name: string; // 秒杀活动名称
  status: number; // 活动状态
  remark: string; // 备注
  start_time: string; // 活动开始时间
  end_time: string; // 活动结束时间
  sort: number; // 排序
  config_ids: string; // 秒杀时段 id 数组
  total_limit_count: number; // 总限购数量
  single_limit_count: number; // 单次限够数量
  stock: number; // 秒杀库存
  total_stock: number; // 秒杀总库存
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
  }

export interface MallPromotionSeckillActivityQueryCondition extends PaginatedRequest {
  
}

export const createMallPromotionSeckillActivity = (mall_promotion_seckill_activity: MallPromotionSeckillActivityRequest): Promise<number> => {
  return api.post<number>(apis.create, mall_promotion_seckill_activity);
}

export const updateMallPromotionSeckillActivity = (mall_promotion_seckill_activity: MallPromotionSeckillActivityRequest): Promise<void> => {
  return api.post<void>(apis.update, mall_promotion_seckill_activity);
}

export const deleteMallPromotionSeckillActivity = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
}

export const getMallPromotionSeckillActivity = (id: number): Promise<MallPromotionSeckillActivityResponse> => {
  return api.get<MallPromotionSeckillActivityResponse>(`${apis.get}/${id}`);
}

export const listMallPromotionSeckillActivity = (): Promise<Array<MallPromotionSeckillActivityResponse>> => {
  return api.get<Array<MallPromotionSeckillActivityResponse>>(apis.list);
}

export const pageMallPromotionSeckillActivity = (condition: MallPromotionSeckillActivityQueryCondition): Promise<PaginatedResponse<MallPromotionSeckillActivityResponse>> => {
  return api.get<PaginatedResponse<MallPromotionSeckillActivityResponse>>(apis.page, condition);
}

export const enableMallPromotionSeckillActivity = (id: number): Promise<void> => {
  return api.post<void>(`${apis.enable}/${id}`);
}

export const disableMallPromotionSeckillActivity = (id: number): Promise<void> => {
  return api.post<void>(`${apis.disable}/${id}`);
}